server/src/instant/admin/routes.clj

```diff
@@ -298,21 +298,34 @@
 
 (defn refresh-tokens-post [req]
   (let [{:keys [app-id]} (req->app-id-authed! req :data/write)
-        email (ex/get-param! req [:body :email] email/coerce)
-        {user-id :id :as user}
-        (or (app-user-model/get-by-email {:app-id app-id
-                                          :email email})
+        email (ex/get-optional-param! req [:body :email] email/coerce)
+        id (ex/get-optional-param! req [:body :id] uuid-util/coerce)
 
-            (app-user-model/create!
-             {:id (UUID/randomUUID)
-              :app-id app-id
-              :email email}))
+        {user-id :id :as user}
+        (cond
+          email (or (app-user-model/get-by-email {:app-id app-id
+                                                  :email email})
+
+                    (app-user-model/create!
+                     {:id (UUID/randomUUID)
+                      :app-id app-id
+                      :email email}))
+          id  (or  (app-user-model/get-by-id {:app-id app-id :id id})
+                   (app-user-model/create!
```

Comment:
I went back and forth about creating a new user based on an `id`. Ultimately went with it, thinking it would be the most consistent with what is happening in the `email` case. Open for other ideas though!